index.js

```diff
@@ -0,0 +1,23 @@
+
+const express = require("express");
+const bodyParser = require("body-parser");
+const xregexp = require('xregexp');
+
+const app = express();
+
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: true }));
+
+app.post('/', function(request, response){
+    referenceData=request.body.referenceData;
+    var transform= JSON.stringify(request.body.payload);
+    for(ref in referenceData)
+        transform=transform.replace(xregexp(`{${ref}}`, 'g'),referenceData[ref]);
```

Comment:
this works for smaller sized request data.
can you check with a larger input file(say 4MB json) and let me know if your code runs without slowing down the system?
if you find it slow, use json object iteration/ walkthrough method to replace placeholders and compare the system load?